docset/winserver2022-ps/failoverclusters/Set-ClusterAffinityRule.md

```diff
@@ -0,0 +1,227 @@
+---
+description: Set-ClusterAffinityRule
+external help file: ClusterAffinityRule.cdxml-help.xml
+Module Name: FailoverClusters
+ms.date: 07/27/2023
+online version: https://learn.microsoft.com/powershell/module/failoverclusters/set-clusteraffinityrule?view=windowsserver2022-ps&wt.mc_id=ps-gethelp
+schema: 2.0.0
+title: Set-ClusterAffinityRule
+---
+
+# Set-ClusterAffinityRule
+
+## SYNOPSIS
+Enabled or Disable an affinity rule, and update the rule type.
+
+## SYNTAX
+
+### Query (cdxml) (Default)
+
+```
+Set-ClusterAffinityRule [[-Name] <String[]>] [-RuleType <RuleType>] [-Enabled <UInt32>]
+ [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-PassThru] [<CommonParameters>]
+```
+
+### InputObject (cdxml)
+
+```
+Set-ClusterAffinityRule -InputObject <CimInstance[]> [-RuleType <RuleType>] [-Enabled <UInt32>]
+ [-CimSession <CimSession[]>] [-ThrottleLimit <Int32>] [-AsJob] [-PassThru] [<CommonParameters>]
+```
+
+## DESCRIPTION
+
+This cmdlet is used to Enabled or Disable an affinity rule. This can also be used to change the
+RuleType.
+
+## EXAMPLES
+
+### Example 1
+
+```powershell
+Set-ClusterAffinityRule -Name AffinityRule1 -Enabled -Cluster Cluster1
+```
+
+This example enables the cluster affinity rule named AffinityRule1.
+
+## PARAMETERS
+
+### -AsJob
+
+Runs the cmdlet as a background job. Use this parameter to run commands that take a long time to
+complete.
+
+The cmdlet immediately returns an object that represents the job and then displays the command
+prompt. You can continue to work in the session while the job completes. To manage the job, use the
+`*-Job` cmdlets. To get the job results, use the
+[Receive-Job](https://go.microsoft.com/fwlink/?LinkID=113372) cmdlet.
+
+For more information about Windows PowerShell background jobs, see
+[about_Jobs](https://go.microsoft.com/fwlink/?LinkID=113251).
+
+```yaml
+Type: SwitchParameter
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -CimSession
+
+Runs the cmdlet in a remote session or on a remote computer. Enter a computer name or a session
+object, such as the output of a [New-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227967)
+or [Get-CimSession](https://go.microsoft.com/fwlink/p/?LinkId=227966) cmdlet. The default is the
+current session on the local computer.
+
+```yaml
+Type: CimSession[]
+Parameter Sets: (All)
+Aliases: Session
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -Enabled
+
+Enables or disables the affinity rule.
+
+```yaml
+Type: UInt32
+Parameter Sets: (All)
+Aliases:
+
+Required: False
+Position: Named
+Default value: None
+Accept pipeline input: False
+Accept wildcard characters: False
+```
+
+### -InputObject
+
+Specifies the input object that is used in a pipeline command.
```

Comment:
Can this be made specific to the cmdlet? It seems like the input object here would be a rule object.